app/controllers/repositories_controller.rb

```diff
@@ -2,14 +2,29 @@ class RepositoriesController < SessionsController
   before_action :current_user
   before_action :signed_in, except: [:index, :show, :download, :download_files]
   before_action :set_repository, only: [:show, :add_like, :destroy, :edit, :update, :download_files]
+  before_action :check_session
+  before_action :check_auth
 
   def show
-    @photos = @repository.photos.first(5)
-    @files = @repository.repo_files.order("LOWER(file_file_name)")
-    @categories = @repository.categories
-    @equipments = @repository.equipments
-    @comments = @repository.comments.order(comment_filter).page params[:page]
-    @vote = @user.upvotes.where(comment_id: @comments.map(&:id)).pluck(:comment_id, :downvote)
+    if @repository.share_type.eql?("private")
+      if @check_passed
+          @photos = @repository.photos.first(5)
+          @files = @repository.repo_files.order("LOWER(file_file_name)")
+          @categories = @repository.categories
+          @equipments = @repository.equipments
+          @comments = @repository.comments.order(comment_filter).page params[:page]
+          @vote = @user.upvotes.where(comment_id: @comments.map(&:id)).pluck(:comment_id, :downvote)
+      else
+          redirect_to password_entry_repository_path
+      end
+    else
+      @photos = @repository.photos.first(5)
+      @files = @repository.repo_files.order("LOWER(file_file_name)")
+      @categories = @repository.categories
+      @equipments = @repository.equipments
+      @comments = @repository.comments.order(comment_filter).page params[:page]
+      @vote = @user.upvotes.where(comment_id: @comments.map(&:id)).pluck(:comment_id, :downvote)
+    end
```

Comment:
You can probably restructure a lot of this so that it has a single early return statement instead of 3 control paths as it is now. Something like

```
if @repository.private? && !check_passed
 redirect_to password_entry_repository_path and return #this last part is the bit that returns early
end

# existing code goes here
```